src/test/java/org/openrewrite/java/testing/hamcrest/RemoveIsMatcherTest.java

```diff
@@ -68,4 +68,50 @@ void testEquals() {
             """));
     }
 
+    @Test
+    void isNotCalledInAssertThat() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.jupiter.api.Test;
+              import static org.hamcrest.Matchers.is;
+              import static org.hamcrest.Matchers.equalTo;
+              
+              class ATest {
+                  @Test
+                  void testMethod() {
+                      String str1 = "Hello world!";
+                      Matcher<String> x = is(equalTo(str1));
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void isNotDirectlyInAssertThat() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.jupiter.api.Test;
+              import static org.hamcrest.MatcherAssert.assertThat;
+              import static org.hamcrest.Matchers.is;
+              import static org.hamcrest.Matchers.equalTo;
+              
+              class ATest {
+                  @Test
+                  void testMethod() {
+                      String str1 = "Hello world!";
+                      String str2 = "Hello world!";
+                      assertThat(str1, contains(is(equalTo(str2))));
```

Comment:
Perhaps `not` is better here as an example than contains, as the latter does not compile.
```suggestion
                      assertThat(str1, not(is(equalTo(str2))));
```